Highway.cpp

```diff
@@ -24,21 +22,27 @@ void Highway::addVehicleInternal(Vehicle* v)
     {
         motorcycle->lanesplitAndRace(100);
     }
+    if(auto* semiTruck = dynamic_cast<SemiTruck*>(v))
+    {
+        semiTruck->driveInSlowLane();
+    }
```

Comment:
You are checking if you can cast for each type, regardless of if a cast succeeds.   That means if `v` is a `Car*`, you're still checking to see if it's a `Motorcycle*` or `SemiTruck*`.  
use `else if` 
- [ ] revise